File: backend/src/main/java/com/yaquedo/service/ContractService.java
package com.yaquedo.service;

import com.yaquedo.dto.ApiResponse;
import com.yaquedo.dto.ContractCreateDto;
import com.yaquedo.dto.ContractDto;
import com.yaquedo.dto.ContractStatusUpdateDto;
import com.yaquedo.dto.PageResponseDto;

public interface ContractService {
    ApiResponse<ContractDto> createContract(String userEmail, ContractCreateDto dto);
    ApiResponse<PageResponseDto<ContractDto>> getMyContracts(String userEmail, int page, int size);
    ApiResponse<ContractDto> updateContractStatus(String userEmail, Long contractId, ContractStatusUpdateDto dto);
}

File: frontend/src/app/services/contract.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  ApiResponse,
  ContractCreateDto,
  ContractDto,
  ContractStatusUpdateDto,
  PageResponseDto,
} from '../models/models';
import { environment } from '../../environments/environment';

const API_BASE = environment.apiUrl;

@Injectable({ providedIn: 'root' })
export class ContractService {
  constructor(private http: HttpClient) {}

  createContract(dto: ContractCreateDto): Observable<ApiResponse<ContractDto>> {
    return this.http.post<ApiResponse<ContractDto>>(`${API_BASE}/contracts`, dto);
  }

  getMyContracts(page = 0, size = 20): Observable<ApiResponse<PageResponseDto<ContractDto>>> {
    return this.http.get<ApiResponse<PageResponseDto<ContractDto>>>(
      `${API_BASE}/contracts/mine?page=${page}&size=${size}`
    );
  }

  cancelContract(contractId: number): Observable<ApiResponse<ContractDto>> {
    return this.http.patch<ApiResponse<ContractDto>>(
      `${API_BASE}/contracts/${contractId}/status`,
      { status: 'CANCELLED' } as ContractStatusUpdateDto
    );
  }
}
